### ISSUE TITLE
1.16.1: Hexcode issues + CMI tabcomplete

### ISSUE THREAD

**Issue by Sikatsu** [kangarko/ChatControl-Pro#1869]
Might be related to my other issue, #1795 but when I use /tell it also shows with the hexcodes.
And since it no longer shows normal minecraft names, people struggle with this a lot on my server now..

![image](https://user-images.githubusercontent.com/5664135/90545621-31c53b80-e189-11ea-8da6-7f9ca64f544e.png)

**Comment by AngellitoX** [comment]
Interesting issue. Have you checked if the issue comes **without** ChatControl?

If so, then probably the issue is coming directly from there, although I would say it might come from here, but it's best to be sure.

**Comment by Sikatsu** [comment]
Hey there! This issue only persists in ChatControl, I have tested this with HyperDrive and CMI alone which worked fine.. it's probably related to my other ticket, where hexcodes don't parse correctly in nicknames for ChatControl

**Comment by AngellitoX** [comment]
Alrighty, let's wait for Matej to give it a look. Thanks for reporting it man!

**Comment by Sikatsu** [comment]
Hey Kangarko,

CMI nicknames still do not parse hexcodes in the private message tab suggestions as shown below.

When I do /w Jarno it says the player is not online, he has a hexcode name but the hexcode doesn't show for him for him.

![image](https://user-images.githubusercontent.com/5664135/91645601-bbf18780-ea46-11ea-8c98-fe6198550ce1.png)

In public chat his nickname is like this: 

![image](https://user-images.githubusercontent.com/5664135/91645644-1985d400-ea47-11ea-8939-2b804f9d78fe.png)

When I use /w AllDudeSons (instead of his nickname) it works but his name is gray. 

![image](https://user-images.githubusercontent.com/5664135/91645655-2a364a00-ea47-11ea-845c-b151162722c9.png)

I am not entirely sure how you test it and if you understand the issues I am experiencing. It's been a bug (partially, since it got worse after CMI implementation) on my server for over a month and players are really annoyed by it.

So,

1) Hexcodes do not parse in tab-completion if it's more then 1. If it's just 1, it parses but you can't use the tab complete as it will say the player is offline.

2) In private messages the hexcodes either don't parse, or return a default minecraft color instead.

If you need any information to debug this further, please let me know

**Comment by kangarko** [comment]
Thank you, sorry this got slipped through my inspection, the changelog for 8.8.0 was quite massive. I'm out of home for the week but I've noted your issue and will return to it as soon as possible, end of this week or the next.

**Comment by Sikatsu** [comment]
Thank you, appreciated.

**Comment by kangarko** [comment]
I discovered that oddly, the matching was case sensitive but broken when the nick contained an uppercase letter, hence it worked for me, but not for you (Janko). That is now resolved!

**Comment by Sikatsu** [comment]
Awesome, and the other hexcode related issues?

**Comment by kangarko** [comment]
Please open a separate ticket for each of those. I noticed some tickets are already open, I will try to patch them one by one.

**Comment by Sikatsu** [comment]
You first told me to do it all in one ticket though..

Will do that later since i'm on holiday. But i've been waiting on a fix for this for 2 months by now :/

**Comment by kangarko** [comment]
I apology for the delay, yes I was aware that it is taking some time however I had to travel unexpectedly for over 10 days.

The reason I requested you open a new ticket next time is because as far as I am not mistaken, the following will get fixed in the next version:

- HEX codes will now work correctly instead of pulling the last letter color
- tab-completing is now fixed and will not include hex colors, you will also be able to private message players with capitalized names or having colors in it.

So we have those two off the list. If you will still have issues with those, simply comment below. If you will experience a new, unrelated problem, it is the best to open a new ticket for clarity.

**Comment by Sikatsu** [comment]
It however still includes hexcodes in tab-completion since the CMI nickname support.

When I just use /w Talia without the hexcodes it doesn't work either.

![image](https://user-images.githubusercontent.com/5664135/92725623-afd1c800-f36c-11ea-843b-a989ea7a4e62.png)

**Comment by kangarko** [comment]
Sorry I cannot reproduce that problem.

- Using 1.16.3 in online mode
- CMI installed and nick changed
- The /w command is not working on default settings since it is not mapped as private message command alias in ChatControl
- Running /tell does tab-complete properly:

![a](https://i.imgur.com/wkuWCFE.png)

Here's my test settings (tested both Bungee true/false and worked)

![b](https://i.imgur.com/eRwdMwb.png)

**Comment by Sikatsu** [comment]
Can you try with more then one hexcode in your nickname?
The issue applies to /reply and /r, not to /tell and it's sub-commands.

I'm using 1.16.3 as well, running on bungee.

**Comment by Sikatsu** [comment]
It seems (for now) that the tab-completion is working correct on 8.8.5, but it still shows wrong nickname colors.

Using /tell and /r results:

![image](https://user-images.githubusercontent.com/5664135/93073168-c89af000-f682-11ea-85ab-f4c23a77b7fd.png)

So it seems in private messages it uses the default minecraft red color instead of the hexcode.

Results of public chat, party chat and ChatControl's private message system:

![image](https://user-images.githubusercontent.com/5664135/93073641-84f4b600-f683-11ea-9052-58233753ab0c.png)

**Comment by Sikatsu** [comment]
Oddly enough still a case where a hexcode shows up... so it must be related to something that it breaks in some cases...

![image](https://user-images.githubusercontent.com/5664135/93075091-99d24900-f685-11ea-96e5-e01f5320ec53.png)

**Comment by Sikatsu** [comment]
![image](https://user-images.githubusercontent.com/5664135/93128258-7fbd5880-f6cf-11ea-81ab-0106a5d11817.png)

**Comment by kangarko** [comment]
Thanks

So I have bad news and good news. Bad news is that while I was gone the code apparently got worse and I cannot manage to fix this problem effectively, good news is that I am rewriting the plugin from the ground up and the new version will come in a few weeks.

Maybe we can do this --> you can use a competitors plugin such as VentureChat for formatting and keep ChatControl for rules until this is resolved?

**Comment by Sikatsu** [comment]
I use quite some features ChatControl has that VentureChat doesn't have, it would be quite some work getting that right. I don't mind waiting an extra max of 2 weeks though, if that fixes all the problems I am experiencing above. It's mostly annoying for the players and in general just looks bad.

I am however confused nobody else reported this issue, perhaps they don't use hexcodes in nicknames..

If you add your own nickname support to ChatControl, could you add a permission so we can allow just certain hexcodes within the nicknames? I only want them to use the hexcodes we have in our ranks, this is currently possible with CMI but I'd love to use ChatControl's nickname system, considering it also works bungee-wide.

Chatcontrol.nick.color.FFFFF to use {#FFFFF} for examplem

**Comment by kangarko** [comment]
Thanks. We had some minor HEX reports but unrelated to this.

Awesome suggestion, will add this ;)

**Comment by kangarko** [comment]
Closing, implemented to CHC 9.1.0 including HEX permissions.
